lib/Installation/Partitioner/FormattingOptionsPage.pm

```diff
@@ -90,18 +94,7 @@ sub select_filesystem {
     wait_screen_change(sub {
             send_key 'end';
     }, 20);
-    if ($filesystem eq 'swap') {
-        send_key_until_needlematch(FILESYSTEM_SWAP, 'up');
-    }
-    elsif ($filesystem eq 'fat') {
-        send_key_until_needlematch(FILESYSTEM_FAT, 'up');
-    }
-    elsif ($filesystem eq 'ext4') {
-        send_key_until_needlematch(FILESYSTEM_EXT4, 'up');
-    }
-    else {
-        die "\"$filesystem\" Filesystem is not known.";
-    }
+    send_key_until_needlematch((sprintf FILESYSTEM_TYPE, $filesystem), 'up');
```

Comment:
Not sure if we need a constant then, as it becomes string template. @OleksandrOrlov WDYT?
From what I can see:
```
 FILESYSTEM_SWAP         => 'partitioning_raid-swap_format-selected',
    FILESYSTEM_FAT          => 'partitioning_raid-fat_format-selected',
    FILESYSTEM_EXT4         => 'partitioning_raid-filesystem_ext4',
```
We then need to add 'partitioning-ext4-format-selected' tag to needles with 'partitioning_raid-filesystem_ext4'
and rename `partitioning_raid-fat_format-selected` -> `partitioning_-fat_format-selected`.
Do you have MR for needles? We need this for openSUSE too.

Reply:
@rwx788, I did not have strict rule on that. Also, I already used the approach with the template stored in a constant, so cannot blame Sofia in using this approach:)

But now, when you pointed to the alternative solution I have to admit, that I would prefer it more. So, using a constant for each needle is more preferable in the case, as it makes the test more transparent and clear.

Reply:
In the other hand handling several needles with the same behavior will require 'if-else' clause. So, not sure what is more evil.

Reply:
It would be typical for perl to check f constant is defined in the runtime and use it if it is:
```
In the rare case in which you need to discover at run time whether a particular constant has been declared via this module, you may use this function to examine the hash %constant::declared . If the given constant name does not include a package name, the current package is used.
``` (https://perldoc.perl.org/constant.html)

And we also can define hash as a constant. Drawback is as before that constants are always available and take space in the memory.

Reply:
I am not sure if I understood. Should I leave it as it was?

Reply:
Sorry for our confusing conversation here:) 

In your solution, the new constant `FILESYSTEM_TYPE` introduces the template `partitioning_%s-format-selected`, but the existing needle tags had a little bit different format. (e.g. in `partitioning_raid-fat_format-selected`. underscore (_) is used before `format`, but in the new template it is a dash (-). `partitioning_raid-filesystem_ext4` has another format at all).

Could you please update needle tags in needles repos (both for osd and o3) to make the solution work.

Reply:
Will do.

Reply:
https://github.com/os-autoinst/os-autoinst-needles-opensuse/pull/640
and 
https://gitlab.suse.de/openqa/os-autoinst-needles-sles/merge_requests/1322

Reply:
Normally we paste this info always in the first comment just before your verification run, it is more visible than here that is quite hidden and maybe nobody will remember to merge them.

Reply:
ok, I edited my initial comment.